UI/ChatGPT/ChatGPT/ChatGPT/Services/ChatService.cs

```diff
@@ -0,0 +1,93 @@
+using ChatGPT.Business;
+using OpenAI.Managers;
+using OpenAI;
+using System.Runtime.CompilerServices;
+using OpenAI.ObjectModels.RequestModels;
+using OpenAI.ObjectModels;
+using System.Text;
+using OpenAI.ObjectModels.ResponseModels;
+
+namespace ChatGPT.Services;
+public class ChatService : IChatService
+{
+	private OpenAIService _client;
+
+	public ChatService(IOptions<AppConfig> appConfig)
+	{
+		var apiKey = appConfig.Value.ApiKey;
+
+		_client = new OpenAIService(
+			new OpenAiOptions()
+			{
+				ApiKey = apiKey
+			});
+	}
+
+	public async ValueTask<ChatResponse> AskAsync(string prompt)
+	{
+
+		var result = await _client.ChatCompletion.CreateCompletion(new ChatCompletionCreateRequest()
+		{
+			Messages = new List<ChatMessage>
+			{
+				ChatMessage.FromUser(prompt)
+			},
+			Model = Models.Gpt_3_5_Turbo
+		});
+
+		if (result.Successful)
+		{
+			var _result = result.Choices.Select(choice => choice.Message.Content);
+			return new ChatResponse(Status.Value, string.Join("", _result));
+		}
+		else
+		{
+			if (result.Error == null)
+			{
+				throw new Exception("Unknown Error");
+			}
+			return new ChatResponse(Status.Error, $"{result.Error.Message ?? "Unknown Error Message"} {result.Error.Code ?? "Unknown Error Code"}");
```

Comment:
String concatenation in business is usually not a good idea as it cannot be localized nor be properly formatted by the UI. You should only return the `result.Error.Message` (and ignore the code for now).

```suggestion
			return new ChatResponse(Status.Error, result.Error.Message);
```

(Note: You might have to update the `ChatResponse.Message` to be a  nullable string)

Reply:
Done in next commits.